src/test/java/com/br/levelup/SubCategoryTest.java

```diff
@@ -0,0 +1,119 @@
+package com.br.levelup;
+
+import com.br.levelup.model.Category;
+import com.br.levelup.model.SubCategory;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.NullAndEmptySource;
+import org.junit.jupiter.params.provider.NullSource;
+
+import java.util.List;
+
+import static com.br.levelup.model.SubCategory.*;
+
+public class SubCategoryTest {
+
+    private Category category = new Category("Programacao", "java");
+
+    @Test
+    void should_build_new_subcategory() {
+        SubCategory subCategory = new SubCategory("Banco de dados", "sql-server", category);
+        Assertions.assertNotNull(subCategory);
+    }
+
+    @ParameterizedTest
+    @NullAndEmptySource
+    void should_throw_exception_when_name_is_invalid(String name) {
+        Assertions.assertThrows(IllegalArgumentException.class,
+                () -> new Category(name, "java-a"));
+    }
+
+    @ParameterizedTest
+    @CsvSource({"Java", "java-*", "java1", "java 1"})
+    void should_throw_exception_when_code_is_invalid(String code) {
+        Assertions.assertThrows(IllegalArgumentException.class,
+                () -> new Category("Programacao", code));
```

Comment:
Não seria para testar o code da SubCategory? poque se for da Category deveria estar no CategoryTest *request change